sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1963,6 +1963,13 @@ object EliminateLimits extends Rule[LogicalPlan] {
     case GlobalLimit(l, child) if canEliminate(l, child) =>
       child
 
+    case Limit(IntegerLiteral(0), child) =>
```

Comment:
It seems we don't need to match `Limit` here since `LocalLimit` and `GlobalLimit` is matched below.

Reply:
Because `transformDownWithPruning` will match here first, then we can replace it with empty `LocalRelation` directly.

Reply:
If the plan is `GlobalLimit(0, LocalLimit(0, child))`, I think this case is useless as we can match `GlobalLimit(0, ...)` directly and turn the plan to empty relation

Reply:
That's why `OptimizeLimitZero` did not match `Limit`

Reply:
> If the plan is `GlobalLimit(0, LocalLimit(0, child))`, I think this case is useless as we can match `GlobalLimit(0, ...)` directly and turn the plan to empty relation

I got it.